File: monkey.cpp
#include <iostream>
using namespace std;
int main()
{
   int n,m,p,k,j;
   cout<<"enter total no. of monkey:";
   cin>>n;
   cout<<"enter total no. of banana & peanut:";
   cin>>m>>p;
   cout<<"enter no. of eatable banana & peanuts:";
   cin>>k>>j;
   if(k<=0 || j<=0)
   {
       cout<<"invalid input:";
   }
   else 
   {
       n-=(m/k)+(p/j);
       if(m%k!=0 && p%j!=0)
       {
           n--;
       }
       if(n<=0)
       {
           cout<<"no. of monkey left on the tree: 0";
       }
       else
       {
           cout<<"no. of monkey left on the tree:"<<n;
       }
   }
}

File: monkey.py
def monkeys_left(n, k, j, m, p):
    total_monkeys = n
    while total_monkeys > 0 and (m >= k or p >= j):
        if m >= k:
            m -= k
        elif p >= j:
            p -= j
        else:
            break
        total_monkeys -= 1 
    return total_monkeys

# Input reading
n = int(input("Enter total monkeys: "))
k = int(input("Enter bananas per monkey: "))
j = int(input("Enter peanuts per monkey: "))
m = int(input("Enter total bananas: "))
p = int(input("Enter total peanuts: "))

# Calculate and print result
result = monkeys_left(n, k, j, m, p)
print(result)
